core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -327,8 +323,8 @@ private static boolean containsNull(Map<Integer, Long> nullValueCounts, Types.Ne
     return nullValueCount > 0;
   }
 
-  private static Stream<DeleteFile> limitBySequenceNumber(
-      long sequenceNumber, long[] seqs, DeleteFile[] files) {
+  private static Stream<IndexedDeleteFile> limitBySequenceNumber(
```

Comment:
Maybe move this into `DeleteFileGroup` to keep the code in one place?

Reply:
Moved.